multiapi-engine/src/main/java/com/sngular/api/generator/plugin/asyncapi/AsyncApiGenerator.java

```diff
@@ -224,25 +232,53 @@ private Map<String, JsonNode> getAllSchemas(final FileLocation ymlParent, final
     final Map<String, JsonNode> totalSchemas = new HashMap<>();
     final List<JsonNode> referenceList = node.findValues(REF);
     referenceList.forEach(reference -> processReference(node, reference, ymlParent, totalSchemas, referenceList));
-    final List<JsonNode> messagesList = node.findValues(MESSAGES);
-    final List<JsonNode> schemasList = node.findValues(SCHEMAS);
-    schemasList.forEach(
-        schema -> schema.fields().forEachRemaining(fieldSchema -> totalSchemas.putIfAbsent((SCHEMAS + SLASH + fieldSchema.getKey()).toUpperCase(), fieldSchema.getValue())));
-    messagesList.forEach(message ->
-                             message.fields().forEachRemaining(fieldSchema -> {
-                               if (fieldSchema.getValue().has(PAYLOAD)) {
-                                 JsonNode payload = fieldSchema.getValue().get(PAYLOAD);
-                                 final String key = (MESSAGES + SLASH + fieldSchema.getKey()).toUpperCase();
-                                 if (payload.has(REF)) {
-                                   payload = payload.get(REF);
-                                 }
-                                 totalSchemas.putIfAbsent(key, payload);
-                               }
-                             })
-    );
+
+    if (node.has(COMPONENTS)) {
+      final List<JsonNode> schemasList = node.get(COMPONENTS).findValues(SCHEMAS);
+      schemasList.forEach(
+          schema -> schema.fields().forEachRemaining(
+              fieldSchema -> totalSchemas.putIfAbsent((SCHEMAS + SLASH + fieldSchema.getKey()).toUpperCase(), fieldSchema.getValue())
+          )
+      );
+
+      final List<JsonNode> messagesList = node.get(COMPONENTS).findValues(MESSAGES);
+      messagesList.forEach(
+          message -> getMessageSchemas(message, totalSchemas)
+      );
+    }
+
+    if (node.has(CHANNELS)) {
+      node.get(CHANNELS).forEach(
+          channel -> getChannelSchemas(channel, totalSchemas)
+      );
+    }
+
     return totalSchemas;
   }
 
+  private void getMessageSchemas(final JsonNode message, final Map<String, JsonNode> totalSchemas) {
+    if (message.has(PAYLOAD)) {
+      JsonNode payload = message.get(PAYLOAD);
+      if (!payload.has(REF)) {
+        final String key = (MESSAGES + SLASH + message.get(NAME).textValue()).toUpperCase();
+        if (payload.has(REF)) {
```

Comment:
Sorry, but here there is something that I don't understand:
If I'm not wrong, L.265 won't be ever reached, because to do it `A and not A` must be `true`
```
if (!payload.has(REF)) {
...
if (payload.has(REF)) {
// un-reachable code
}
...
}
```
Please, tell me if I'm wrong.

Reply:
Yep, absolutely. Changed some of the logic and missed that